src/emulate.rs

```diff
@@ -815,6 +832,18 @@ fn exec_div(state: &mut EmulatorState, rtype: RType) {
     state.pc_next();
 }
 
+// rd = s64(rs1{32} /s rs2{32})
+// pc = pc + 4
+fn exec_divw(state: &mut EmulatorState, rtype: RType) {
+    let rs1_value = state.get_reg(rtype.rs1());
+    let rs2_value = state.get_reg(rtype.rs2());
+    assert!(rs2_value != 0, "check for non-zero divisor");
```

Comment:
I think this assertion will not catch the case where some of the upper 32 bits in rs2 are non-zero. Better to use `assert!((rs2_value as i32) != 0, ...)` instead.